distributed/client.py

```diff
@@ -2936,8 +2936,18 @@ def compute(
         if not isinstance(priority, Number):
             priority = {k: p for c, p in priority.items() for k in self._expand_key(c)}
 
+        if not isinstance(dsk, HighLevelGraph):
+            dsk = HighLevelGraph.from_collections(id(dsk), dsk, dependencies=())
+
+        # Let's append the finalize graph to dsk
+        layers = {f"finalize-{id(dsk2)}": dsk2}
```

Comment:
We might want to use a uuid here instead.  `id` is known to produce duplicate values in practice because dsk2 might be released and then another object might use the same object ID.

Reply:
Or maybe `tokenize(names)` would be better

Reply:
Good point, fixed